fetch/api/request/destination/resources/fetch-destination-worker-iframe.js

```diff
@@ -0,0 +1,22 @@
+self.addEventListener('fetch', function(event) {
+    if (event.request.url.includes('dummy')) {
+        event.waitUntil(async function() {
+            let destination = new URL(event.request.url).searchParams.get("dest");
+            await self.clients.matchAll({"includeUncontrolled": true})
+                .then(function(clients) {
```

Comment:
nit: You can avoid the `then()` and its nesting since you are in an async function.  Something like:

```javascript
let clients = await self.clients.matchAll({includeUncontrolled: true});
clients.forEach(...)
```

Reply:
good call :)